### PR TITLE
Add FromStr trait for Key

### PR DESCRIPTION
**Pull request by farmaazon**
Extended the `convert.py` script, so it implement trait `FromStr` for Key. Thanks to that, one can now parse strings from e.g. `js_sys::KeyboardEvent::key()` and have the enum value.

### CODE REVIEW

File: convert.py [hunk 1]
```diff
@@ -58,7 +62,31 @@ def convert_key(text, file):
...
+    print_from_str_entries(display, file)
+    print("""
+            _ => Err(UnrecognizedKeyError),
+        }
+    }
+}
+
+/// Parse from string error, returned when string does not match to any Key variant.
+#[derive(Clone, Debug)]
+pub struct UnrecognizedKeyError;
```
Comment (pyfisch) [inline convert.py]: Please implement the std error trait for this one.

File: src/key.rs [hunk 2]
```diff
@@ -915,6 +916,313 @@ impl Display for Key {
...
-} 
+}
+
+impl FromStr for Key {
+    type Err = UnrecognizedKeyError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        use Key::*;
+        match s {
+            ch if ch.chars().take(2).count() == 1 => Ok(Character(ch.to_string())),
```
Comment (pyfisch) [inline src/key.rs]: A key string can also consist of a base character and multiple combining characters.
https://w3c.github.io/uievents-key/#keys-unicode
Not sure if it is worth supporting.

Additionally Control Characters should be excluded.

Comment (farmaazon) [reply src/key.rs]: I could add the exact check if string is a _key string_, but I'm not sure how do you fell with adding new dependency (I would use https://docs.rs/unic-ucd-category/0.9.0/unic_ucd_category/enum.GeneralCategory.html )

Comment (pyfisch) [reply src/key.rs]: Yeah, the dependency would be quite big for such a small feature.

How about checking that:

1. there are no control characters in the string (`char::is_control`)
2. No characters besides the first are in the ASCII range (`char::is_ascii`)

This prevents common mistakes like mistyped keynames (e.g. `Ennter`) from being recognized as characters while allowing all characters specified. If the need arises we can add the dependency later.

**Comment by pyfisch** [comment]
Released as v0.5.0

**Pull request merged**